### PR TITLE
Feature estimate gas

### LINKED ISSUE #345: if eth_estimateGasLimit fails, we must return the block gas limit
Analyse the eth_estimateGasLimit error, and if it is not a legitimate contracts error, then default to the block gas limit.

- [ ] RPC issue (timeout)

### PR DESCRIPTION
**Pull request by Yashk767**
# Description

Added condition to send max gasLimit when the estimateGas throws an error which is an rpc error or too low intrinsic gas error

Fixes #345

### CODE REVIEW

File: utils/options.go [hunk 1]
```diff
@@ -44,6 +44,15 @@ func (*UtilsStruct) GetTxnOpts(transactionData types.TransactionOptions) *bind.T
 
 	gasLimit, err := UtilsInterface.GetGasLimit(transactionData, txnOpts)
 	if err != nil {
+		errString := err.Error()
+		if ContainsStringFromArray(errString, []string{"500", "501", "502", "503", "504"}) || errString == errors.New("intrinsic gas too low").Error() {
+			latestBlock, err := UtilsInterface.GetLatestBlockWithRetry(transactionData.Client)
```
Comment (ashish10677) [inline utils/options.go:49]: Can you put a debug log saying why we are sending block gas limit

**Review by ashish10677 — CHANGES_REQUESTED** [review]

File: utils/array.go [hunk 2]
```diff
@@ -83,3 +84,12 @@ func ConvertUintArrayToUint16Array(uintArr []uint) []uint16 {
 	}
 	return arr
 }
+
+func ContainsStringFromArray(source string, subStringArray []string) bool {
```
[the remark this answers is not in the document]
Comment (Yashk767) [reply utils/array.go:88]: But we are checking for substring here, the existing one will check for the entire string right ?

[the remark this answers is not in the document]
Comment (Yashk767) [reply utils/array.go:88]: Here we are checking whether any string from array is a sub-string of source string

[the remark this answers is not in the document]
Comment (ashish10677) [reply utils/array.go:88]: Okay got it

**Review by ashish10677 — APPROVED** [review]

**Pull request merged**